### PR TITLE
ISSD-1633[Maketplace] Implementation of DXP and Cloud tags in DPT

### PR DESCRIPTION
**Pull request by juliocezar96**
Subtask: [ISSD-1633](https://liferay.atlassian.net/browse/ISSD-1633)

### CODE REVIEW

**Comment by juliocezar96** [comment]
## :fire: GitRay Report 
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.
 ## Jira Tasks 

 <html></html> 

---

Hey, don't forget to update all related tickets with correct status, such as **In Review** :wink: 
### Git Stats 

4 files changed, 134 insertions(+), 7 deletions(-)

**Comment by juliocezar96** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 1]
```diff
@@ -1,7 +1,122 @@
...
+		}
+	}
+
+</style>
+
+<#assign
+	PRODUCT_TYPE_FREE = "FREE"
+	PRODUCT_TYPE_PAID = "PAID"
+	PRODUCT_TYPE_DXP = "DXP"
+	PRODUCT_TYPE_CLOUD = "CLOUD"
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:41]: Corrigir ordenação.

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 2]
```diff
@@ -1,7 +1,122 @@
...
+
+<#if themeDisplay?has_content>
+	<#assign scopeGroupId = themeDisplay.getScopeGroupId() />
+</#if>
+
+<#if currentURL?has_content>
+	<#if currentURL?contains('web')>
+		<#assign
+			partsUrl = currentURL?split('/')
+			index = 2
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:53]: Corrigir ordenação.

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 3]
```diff
@@ -1,7 +1,122 @@
...
+</#if>
+
+<#if (CPDefinition_cProductId.getData())??>
+	<#assign productId = CPDefinition_cProductId.getData() />
+</#if>
+
+<#assign
+	product = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/"+ channelId +"/products/"+ productId +"?accountId=-1&nestedFields=productSpecifications,categories")
+	productSpecifications = product.productSpecifications![]
+	categories = product.categories![]
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:72]: Corrigir ordenação.

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 4]
```diff
@@ -1,7 +1,122 @@
...
+				<#assign
+					type = ""
+					icon =""
+				/>
+
+				<#if productType.value?upper_case == PRODUCT_TYPE_DXP>
+					<#assign
+						type = "DXP App"
+						icon ="ir-icon-dxp-svg"
+					/>
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:101]: Tem algum caso que teremos mesmo o type e icon definido como vazio?

Corrigir ordenação e espaços.

**Comment by juliocezar96** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 5]
```diff
@@ -1,7 +1,113 @@
...
+			<#list productTypes as productType>
+				<#if productType.value?upper_case == PRODUCT_TYPE_DXP>
+					<#assign
+						type = "DXP App"
+						icon ="ir-icon-dxp-svg"
+					/>
+				<#elseif productType.value?upper_case == PRODUCT_TYPE_CLOUD>
+					<#assign
+						type = "Cloud App"
+						icon = "ir-icon-cloud-svg"
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:98]: Corrigir a ordenação.

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 6]
```diff
@@ -1,7 +1,113 @@
...
+</#if>
+
+<#if (CPDefinition_cProductId.getData())??>
+	<#assign productId = CPDefinition_cProductId.getData() />
+</#if>
+
+<#assign
+	categories = product.categories![]
+	product = restClient.get("/headless-commerce-delivery-catalog/v1.0/channels/"+ channelId +"/products/"+ productId +"?accountId=-1&nestedFields=productSpecifications,categories")
+	productSpecifications = product.productSpecifications![]
```
Comment (eliasantosf) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:70]: Categories precisa vir após o product.

**Review by eliasantosf — CHANGES_REQUESTED** [review]

**Comment by juliocezar96** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 7]
```diff
@@ -1,7 +1,114 @@
...
+
+	.app-container .app-category {
+		background-color: #e6ebf5;
+		padding: 6px 8px 4px 8px;
+	}
+
+	.app-container .app-product-type {
+		border-color: #2e5aac !important;
+		color:#2e5aac;
+		padding: 4px 8px 4px 8px;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:14]:
```suggestion
		padding: 4px 8px;
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 8]
```diff
@@ -1,7 +1,114 @@
...
-			${AssetVocabulary_7784259.getData()}
-		</div>
+<style>
+	.app-container {
+		font-size: MEDIUM;
+	}
+
+	.app-container .app-category {
+		background-color: #e6ebf5;
+		padding: 6px 8px 4px 8px;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:8]:
```suggestion
		padding: 6px 8px 4px;
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 9]
```diff
@@ -1,7 +1,114 @@
...
+	}
+
+	@media screen and (max-width: 576px) {
+		.app-container {
+			font-size: x-small;
+		}
+
+		.app-container .app-category,
+		.app-container .app-product-type {
+			padding: 2px 4px 2px 4px;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:30]:
```suggestion
			padding: 2px 4px;
```

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl [hunk 10]
```diff
@@ -1,7 +1,114 @@
-<#if (AssetVocabulary_7784259.getData())??>
-	<div class="color-neutral-3 flex flex-wrap font-size-paragraph-small solution-category-container">
-		<div class="bg-neutral-8 border-radius-small mb-1 mr-1 px-1 solution-category">
-			${AssetVocabulary_7784259.getData()}
-		</div>
+<style>
+	.app-container {
+		font-size: MEDIUM;
```
Comment (kevenleone) [inline modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:3]: Could we specify the size? medium could be anything

```suggestion
		font-size: MEDIUM;
```

**Review by kevenleone — CHANGES_REQUESTED** [review]
@juliocezar96 can you take a look?

File: modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl (hunk 10 above)
Comment (juliocezar96) [reply modules/apps/site-initializer/site-initializer-liferay-marketplace/src/main/resources/site-initializer/ddm-templates/marketplace-apps-categories/ddm-template.ftl:3]: I can remove it. It doesn't affect anything, I just got it from figma

**Comment by juliocezar96** [comment]
ci:test:sf

**Review by kevenleone — APPROVED** [review]
LGTM 🔥 

Good job @juliocezar96

**Comment by kevenleone** [comment]
ci:test:sf

**Review by eliasantosf — APPROVED** [review]

**Comment by kevenleone** [comment]
ci:forward

**Comment by juliocezar96** [comment]
resent #143930

**Comment by juliocezar96** [comment]
resent #https://github.com/brianchandotcom/liferay-portal/pull/143930

**Comment by juliocezar96** [comment]
resent https://github.com/brianchandotcom/liferay-portal/pull/143965

**Pull request closed without merging**